src/solvers/dgsem_p4est/containers.jl

```diff
@@ -6,14 +6,14 @@
 #! format: noindent
 
 mutable struct P4estElementContainer{NDIMS, RealT <: Real, uEltype <: Real, NDIMSP1,
-                                     NDIMSP2, NDIMSP3} <: AbstractContainer
+                                     NDIMSP2, NDIMSP3, R, S, X, O} <: AbstractContainer
     # Physical coordinates at each node
     node_coordinates::Array{RealT, NDIMSP2}   # [orientation, node_i, node_j, node_k, element]
     # Jacobian matrix of the transformation
     # [jacobian_i, jacobian_j, node_i, node_j, node_k, element] where jacobian_i is the first index of the Jacobian matrix,...
     jacobian_matrix::Array{RealT, NDIMSP3}
     # Contravariant vectors, scaled by J, in Kopriva's blue book called Ja^i_n (i index, n dimension)
-    contravariant_vectors::PtrArray{RealT, NDIMSP3}   # [dimension, index, node_i, node_j, node_k, element]
+    contravariant_vectors::PtrArray{RealT, NDIMSP3, R, S, X, O}   # [dimension, index, node_i, node_j, node_k, element]
```

Comment:
I would use a type parameter such as `ContravariantVectors <: AbstractArray{RealT, NDIMSP3}` or so instead. This would avoid introducing unknown parameters `R, S, X, O` or breaking if their type parameters change.

Reply:
Makes sense! done in 500db95